### PR TITLE
Support setting field title formatter

### PR DESCRIPTION
**Pull request by kanitw**

### CODE REVIEW

File: site/usage/compile.md [hunk 1]
```diff
@@ -24,11 +24,12 @@ If provided, the `options` argument should be an object with one or more of the
 
 * [`config`](#config) sets a default config
 * [`logger`](#logging) sets a logger
+* ['fieldTitle`](#field-title) sets a field title formatter
 
 {:#config}
 ### Customized Configuration
 
-You can specify a [config]({{site.baseurl}}/docs/config.html) object, which is used as the default config. Configurations in the Vega-Lite specification itself override what is passed in through the `compile` function.
+You can specify a [config]({{site.baseurl}}/docs/config.html) object, which is used as the default config, as a property of the `compile` function's `options` argument. Configurations in the Vega-Lite specification itself override what is passed in through the `compile` function.
```
Comment (domoritz) [inline site/usage/compile.md]: This sentence needs to be restructured. It doesn't make sense anymore.

File: site/usage/compile.md [hunk 2]
```diff
@@ -50,6 +51,23 @@ interface LoggerInterface {
 }
 ```
 
+### Customized Field Title Formatter
+
+To customize how Vega-Lite generates axis or legend title for a [field definition](encoding.html#field-def), you can provide a `titleFormat` function as a property of the `compile` function's `options` argument.
```
Comment (domoritz) [inline site/usage/compile.md]: Axis and legend titleS

File: src/compile/compile.ts [hunk 3]
```diff
@@ -8,15 +9,27 @@ import {assembleRootData} from './data/assemble';
 import {optimizeDataflow} from './data/optimize';
 import {Model} from './model';
 
+export interface CompileOption {
```
Comment (domoritz) [inline src/compile/compile.ts]: CompileOptions

Comment (kanitw) [reply src/compile/compile.ts]: Well, our config object isn't `Configs` either.

Comment (domoritz) [reply src/compile/compile.ts]: Because it's one config with may options.

Comment (domoritz) [reply src/compile/compile.ts]: You have a tendency to try to make things consistent when English is the most inconsistent language, lol.

File: site/usage/compile.md [hunk 4]
```diff
@@ -24,11 +24,12 @@ If provided, the `options` argument should be an object with one or more of the
 
 * [`config`](#config) sets a default config
 * [`logger`](#logging) sets a logger
+* ['fieldTitle`](#field-title) sets a field title formatter
 
 {:#config}
 ### Customized Configuration
 
-You can specify a [config]({{site.baseurl}}/docs/config.html) object, which is used as the default config. Configurations in the Vega-Lite specification itself override what is passed in through the `compile` function.
+You can specify a [config]({{site.baseurl}}/docs/config.html) object as a property of the `compile` function's `options` argument. This `config` property will be used as the default configurations. Configurations in the Vega-Lite specification will override the `config` passed in through the `compile` function.
```
Comment (domoritz) [inline site/usage/compile.md]: "The `config` property will be used as the default configuration.", lol

Comment (domoritz) [reply site/usage/compile.md]: "The configuration in the Vega-Lite specification will override the `config` passed in through the `compile` function."

File: src/compile/compile.ts (hunk 3 above)
Comment (kanitw) [reply src/compile/compile.ts]: Whatever plurality is annoying anyway.

**Pull request merged**